### ISSUE TITLE
Order Details: Add Dynamic Type support in Accessibility sizes on buttons

### ISSUE THREAD

**Issue by ealeksandrov** [woocommerce/woocommerce-ios#5335]
### Description
"Add customer note" and "Add shipping address" buttons does not support Dynamic Type.
Needs more investigation/view structure re-arrangement.

### Why
Even when `UIButton`'s title is setup to wrap, superview does not expand its frame for dynamic content.

Possible fix is to use  `enableMultipleLines()`, but the helper won't work properly when button has image (we have "plus" icon in this case):

https://github.com/woocommerce/woocommerce-ios/blob/8b1e845136c94c1c583dead82b34f94600826eba/WooCommerce/Classes/Extensions/UIButton%2BHelpers.swift#L137-L144

## Screenshots

<img width=415 src="https://user-images.githubusercontent.com/3132438/140091461-7f85a2d0-587f-4c06-99e7-71b030629573.png">

**Comment by joshheald** [comment]
Automated triage note: this was tested and commented by Codex using device/simulator automation. Tested on iPad Pro 11-inch (M4) simulator, iOS 18.6, WooCommerce iOS 24.9.0.4 on trunk df01963e5080f79ae419981ce6dc9630d291b484 with Dynamic Type set to accessibility-extra-extra-extra-large. I did not reproduce the button-specific issue: visible Order Details rows such as Add Shipping Address, View Billing Information, and Add Tracking scaled/wrapped and stayed readable. I did notice separate large-text rough edges in non-button rows, but the reported button Dynamic Type failure was not reproduced on this build.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
